### ISSUE TITLE
Can't multiply real parts of two complex matrices

### ISSUE THREAD

**Issue by Saluev** [numpy/numpy#13232]
When I generate two complex matrices of particular size and try to multiply their real parts, numpy gets stuck.

### Reproducing code example:

```python
import numpy as np
A = np.zeros((2048, 4096), dtype=np.complex128)
B = np.zeros((4096, 2048), dtype=np.complex128)
A.real @ B.real
# Expected to run in few seconds, but runs forever. ^C doesn't work.
```

### Numpy/Python version information:

1.16.2 3.6.1 (default, Apr  4 2017, 09:40:51) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]

**Comment by RuchikaChugh** [comment]
I would like to work on this issue.

**Comment by mattip** [comment]
This is a limitation of `matmul` [giving up](https://github.com/numpy/numpy/blob/v1.16.2/numpy/core/src/umath/matmul.c.src#L366) and not using BLAS when doing matrix-matix matmul if any of the input or output are not contiguous. We should be copying the input to temporary buffers, since [`gemm`](http://www.netlib.org/lapack/explore-html/db/dc9/group__single__blas__level3_gafe51bacb54592ff5de056acabd83c260.html#gafe51bacb54592ff5de056acabd83c260) requires that the operands be contiguous along one axis. `'linalg` does the copying via scratch buffers at the beginning of the inner loop.

**Comment by mhvk** [comment]
Indeed, just takes very long as 2^46 operations are done... Though for me the difference between adding `.real` and just doing `A @ B` is not that large (30s vs 11s)-- are we sure this is the same issue?
